nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -164,19 +164,25 @@ def _build_tokenizer(self):
         All tokenizers are expected to provide compatible interface.
         Override default Encoder-decoder tokenizer to use legacy=True for sentencepiece.
         """
-        if hasattr(self._cfg.tokenizer, "sentencepiece_legacy"):
-            legacy = self._cfg.tokenizer.sentencepiece_legacy
+
+        if self._cfg.tokenizer.library == 'music':
+            self.tokenizer = get_tokenizer(
+                tokenizer_name=self._cfg.tokenizer.library, vocab_file=self._cfg.tokenizer.vocab_file
+            )
         else:
-            legacy = True if self._cfg.tokenizer.library == 'sentencepiece' else False
-        self.tokenizer = get_nmt_tokenizer(
-            library=self._cfg.tokenizer.library,
-            model_name=self._cfg.tokenizer.type,
-            tokenizer_model=self.register_artifact("tokenizer.model", self._cfg.tokenizer.model),
-            vocab_file=self.register_artifact("tokenizer.vocab_file", self._cfg.tokenizer.vocab_file),
-            merges_file=self.register_artifact("tokenizer.merge_file", self._cfg.tokenizer.merge_file),
-            delimiter=self.cfg.tokenizer.get('delimiter', None),
-            legacy=legacy,
-        )
+            if hasattr(self._cfg.tokenizer, "sentencepiece_legacy"):
+                legacy = self._cfg.tokenizer.sentencepiece_legacy
+            else:
+                legacy = True if self._cfg.tokenizer.library == 'sentencepiece' else False
+            self.tokenizer = get_nmt_tokenizer(
```

Comment:
get_nmt_tokenizer can retrun the music tokenizer if your set the `cfg.tokenizer.library` correctly. No need to modifying this base class.